crates/qt-build-utils/src/platform.rs

```diff
@@ -0,0 +1,75 @@
+// SPDX-FileCopyrightText: 2025 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use std::{env, process::Command};
+
+fn command_help_output(command: &str) -> std::io::Result<std::process::Output> {
+    Command::new(command).args(["--help"]).output()
+}
+
+/// Linking executables (including tests) with Cargo that link to Qt fails to link with GNU ld.bfd,
+/// which is the default on most Linux distributions, so use GNU ld.gold, lld, or mold instead.
+/// If you are using a C++ build system such as CMake to do the final link of the executable, you do
+/// not need to call this function.
+///
+/// With Apple devices we set -fapple-link-rtlib as we build with -nodefaultlibs
+/// otherwise we cannot user helpers from the compiler runtime in Qt
+///
+/// This does nothing on non-Unix platforms.
+pub struct QtPlatformLinker;
+
+impl QtPlatformLinker {
+    /// Initialize support for linking executables with Cargo that link to Qt
+    pub fn init() {
```

Comment:
don't know if we want to refactor this code at all? Or if there is anything else we can setup for any platforms beyond just the linker. Most of the Windows craziness was on the CMake side?

We do have the cc build flags we setup, they are currently done in cxx-qt-build don't know if we have a helper in qt-build-utils or not. As could be like `QtPlatform::cc_builder() -> cc::Build` with the flags preconfigured